### PR TITLE
Add cosign workflow template.

### PR DESCRIPTION
**Pull request by todaywasawesome**
Signed-off-by: Dan Garfield <dan@codefresh.io>

### CODE REVIEW

File: workflows/cosign/versions/0.0.1/docs/sign.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# task
```
Comment (eti-codefresh) [inline workflows/cosign/versions/0.0.1/docs/sign.md:1]: you need to change the name of the file sign.md to cosign.md 
it should be as the template name 
also change the names of the files cosign.workfolw... and cosign.rbac.. just trim the cosign

File: workflows/cosign/versions/0.0.1/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            path: cosign.password
+          - key: cosign.pub
+            path: cosign.pub
+        secretName: cosignkey
+```
+Make sure `secretName` matches the name of the secret you generated in step 2.
+
+## Templates
+
+1. [Sign](https://github.com/codefresh-io/argo-hub/blob/main/workflows/cosign/versions/0.0.1/docs/sign.md)
```
Comment (eti-codefresh) [inline workflows/cosign/versions/0.0.1/README.md:30]: change this accordingly to cosign

File: workflows/cosign/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+## v0.0.1 (?)
+
+### task
+
+short description
```
Comment (eti-codefresh) [inline workflows/cosign/CHANGELOG.md:7]: maybe add some description

File: workflows/cosign/versions/0.0.1/cosign.workflow-template.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        annotations:
+          argo-hub-template/description: 'Cosign cli image provided by bitnami'
+          argo-hub-template/icon_url: "https://avatars.githubusercontent.com/u/71096353?s=200&v=4"
+          argo-hub-template/icon_background: "#f4f4f4"
+      inputs:
+        parameters:
+          - name: IMAGE_NAME
+          - name: TAG
+            default: latest
+          - name: COSIGN_PASSWORD # Need to find better way to pass this
```
Comment (eti-codefresh) [inline workflows/cosign/versions/0.0.1/cosign.workflow-template.yaml:30]: are you going to leave this comment ?

**Comment by todaywasawesome** [comment]
Thanks @eti-codefresh I made the requested changes.

File: workflows/cosign/versions/0.0.1/docs/cosign.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# task
+
+## Summary
+Create and upload a signature for image.
+
+## Inputs/Outputs
+
+### Inputs
+IMAGE_NAME (required) - Specify the name of the image to be signed, example `example/image`
+TAG (required) - Specify the tag to be signed, example `latest`
```
Comment (eti-codefresh) [inline workflows/cosign/versions/0.0.1/docs/cosign.md:10]: i think that need to add this here also COSIGN_PASSWORD

File: workflows/cosign/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Changelog
+
+## v0.0.1 (?)
```
Comment (eti-codefresh) [inline workflows/cosign/CHANGELOG.md:3]: instead of '?' you can put date

**Review by eti-codefresh — APPROVED** [review]

**Comment by todaywasawesome** [comment]
@eti-codefresh Fixed!

**Review by eti-codefresh — APPROVED** [review]

**Comment by todaywasawesome** [comment]
Closing this PR in favor of https://github.com/codefresh-io/argo-hub/pull/432

**Pull request closed without merging**